### PR TITLE
Establish connection with fhir

### PR DESCRIPTION
**Pull request by thanakritTW**
~~PS. PLEASE MAKE SURE TO SQUASH AND DELETE THIS BRANCH before merging into master as it contains default username and password in the history~~

@praveenadayanand confirms that it is okay to have plaintext url and default creds in the repo

### CODE REVIEW

**Comment by thanakritTW** [comment]
@davidviktorTW could you help review the last 2 commits of dependencies injection

**Comment by davidviktorTW** [comment]
> @davidviktorTW could you help review the last 2 commits of dependencies injection

What you added looks good, but I think the HttpClient is still missing that needs to be injected to OpenMrsClient.

File: src/In.ProjectEKA.HipService/Startup.cs [hunk 1]
```diff
@@ -104,6 +105,9 @@ public Startup(IConfiguration configuration)
                 .AddSingleton(Configuration.GetSection("Gateway").Get<GatewayConfiguration>())
                 .AddSingleton(new GatewayClient(HttpClient,
                     Configuration.GetSection("Gateway").Get<GatewayConfiguration>()))
+                .AddSingleton(Configuration.GetSection("OpenMrs").Get<OpenMrsConfiguration>())
+                .AddSingleton(new OpenMrsClient(HttpClient,
```
Comment (thanakritTW) [inline src/In.ProjectEKA.HipService/Startup.cs:109]: > > @davidviktorTW could you help review the last 2 commits of dependencies injection
> 
> What you added looks good, but I think the HttpClient is still missing that needs to be injected to OpenMrsClient.

isn't it happening in this line just like the GatewayHttpClient

Comment (thanakritTW) [reply src/In.ProjectEKA.HipService/Startup.cs:109]: because this HttpClient is a property of this class 

```
 private HttpClient HttpClient { get; }
```

Comment (davidviktorTW) [reply src/In.ProjectEKA.HipService/Startup.cs:109]: Yes, sorry, you are right!
One minor thing with that (does not need any action in this PR I think) that one is using a timeout setting for the Gateway, which may or may not be the same ideally chosen for the OpenMRS connection.

Comment (thanakritTW) [reply src/In.ProjectEKA.HipService/Startup.cs:109]: I totally agreed

**Pull request merged**